### PR TITLE
Notify users when they get assigned as a group organizer

### PR DESCRIPTION
**Pull request by s-cheng**
This PR adds a new notification type, that gets triggered when a user is added as a group organizer. It includes a migration to set the default value (getting notified both on-site and via email) for all users.

Screen Shot 2022-03-31 at 7 47 33 PM

### CODE REVIEW

File: packages/lesswrong/server/migrations/2022-03-31-defaultGroupAdminNotificationFill.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  name: "defaultGroupAdminNotificationFill",
+  dateWritten: "2022-03-31",
+  idempotent: true,
+  action: async () => {
+    await fillDefaultValues({
+      collection: Users,
+      fieldName: "notificationGroupAdministration",
+    });
+  },
+});
```
Comment (jpaddison3) [inline packages/lesswrong/server/migrations/2022-03-31-defaultGroupAdminNotificationFill.ts:15]: Perfect, glad you found the right migration to copy.

File: packages/lesswrong/server/callbacks/localgroupCallbacks.ts [hunk 2]
```diff
@@ -14,3 +16,9 @@ getCollectionHooks("Localgroups").updateValidate.add((validationErrors: Array<an
   
   return validationErrors;
 });
+
+getCollectionHooks("Localgroups").updateAfter.add(async (newDoc: DbLocalgroup, {oldDocument}: {oldDocument: DbLocalgroup}) => {
+  const newOrganizerIds = difference(newDoc.organizerIds, oldDocument.organizerIds)
+  await createNotifications({userIds: newOrganizerIds, notificationType: "newGroupOrganizer", documentType: "localgroup", documentId: newDoc._id})
+  return newDoc
+})
```
Comment (jpaddison3) [inline packages/lesswrong/server/callbacks/localgroupCallbacks.ts:24]: What about after create?

File: packages/lesswrong/server/callbacks/localgroupCallbacks.ts [hunk 3]
```diff
@@ -14,3 +16,9 @@ getCollectionHooks("Localgroups").updateValidate.add((validationErrors: Array<an
   
   return validationErrors;
 });
+
+getCollectionHooks("Localgroups").updateAfter.add(async (newDoc: DbLocalgroup, {oldDocument}: {oldDocument: DbLocalgroup}) => {
```
Comment (jpaddison3) [inline packages/lesswrong/server/callbacks/localgroupCallbacks.ts:20]: Q: Should this be updateAsync? We don't need to hold up the return of the new document on this.

File: packages/lesswrong/server/notificationTypesServer.tsx [hunk 4]
```diff
@@ -383,3 +385,31 @@ export const NewRSVPNotification = serverRegisterNotificationType({
...
+    
+    return <div>
+      <p>
+        Hi {user.displayName},
+      </p>
+      <p>
+        You've been assigned as a group organizer for <a href={groupLink}>{localGroup.name}</a> on {siteNameWithArticleSetting.get()}.
+      </p>
+      <p>
+        - {capitalize(siteNameWithArticleSetting.get())} Team
```
Comment (jpaddison3) [inline packages/lesswrong/server/notificationTypesServer.tsx:411]: I think you want to say The LessWrong Team here. siteNameWithArticleSetting is designed to allow you to switch between "on LessWrong" and "on the EA Forum". You can just use the regular site name and supply the article.

File: packages/lesswrong/server/notificationTypesServer.tsx [hunk 5]
```diff
@@ -383,3 +385,31 @@ export const NewRSVPNotification = serverRegisterNotificationType({
...
+    if (!localGroup) throw new Error("Cannot find local group for which this notification is being sent")
+    
+    const groupLink = `${getSiteUrl().slice(0,-1)}/groups/${localGroup._id}`
+    
+    return <div>
+      <p>
+        Hi {user.displayName},
+      </p>
+      <p>
+        You've been assigned as a group organizer for <a href={groupLink}>{localGroup.name}</a> on {siteNameWithArticleSetting.get()}.
```
Comment (jpaddison3) [inline packages/lesswrong/server/notificationTypesServer.tsx:408]: Maybe write a description of what they can do?

**Review by jpaddison3 — APPROVED** [review]
LGTM

File: packages/lesswrong/server/callbacks/localgroupCallbacks.ts (hunk 2 above)
Comment (s-cheng) [reply packages/lesswrong/server/callbacks/localgroupCallbacks.ts:27]: Thanks for the comments! Guess this PR was a bit sloppy :P

**Pull request merged**